src/solver/solve.jl

```diff
@@ -6,23 +6,78 @@ struct EfficiencyStats{TS <: Tuple, WT}
     walltime::WT
 end
 
+mutable struct GracefulExit
```

Comment:
Why do we need a mutable struct?

Reply:
Ah, that’s an artifact of my first implementation. I’ll make it immutable